### DISCUSSION TITLE
No <title> tag in the source code despite SSR = no indexing by Bing

### DISCUSSION THREAD

**Discussion by yannick-leguennec** [vercel/next.js#63285] [Help] [answered]
### Summary

I'm encountering an issue where the <title> tag set within the Head component of my Next.js application does not appear in the source HTML when viewed in production mode. Despite using getServerSideProps for Server-Side Rendering (SSR) and ensuring the Head component correctly includes a static <title> tag, the tag is missing in the initial HTML source served to the browser. This behavior persists even after verifying that there are no errors during the build process and simplifying the page structure The absence of the <title> tag in the source HTML impacts SEO and is particularly problematic as search engine crawlers may not execute JavaScript (Bing), relying on the initial HTML content. Testing in both development and production environments has not resolved the issue.

Any ideas?

### Example

wiseduckdev,vercel.app

**Comment by icyJoseph** [comment, marked as answer]
Hi,

Could it be that the title tag is rendered conditionally? So the code path that should render it is not hit until after the client side has run?

This happens typically when one has an `_app` file, that skips rendering the Next.js Component tag, under certain conditions (your business logic).

For example, normally we'd have:

```tsx
import type { AppProps } from 'next/app'
 
export default function MyApp({ Component, pageProps }: AppProps) {
  return <Component {...pageProps} />
}
```

But maybe you'd have something like:

```tsx
import type { AppProps } from 'next/app'
import { useState, useEffect } from 'react';
 
export default function MyApp({ Component, pageProps }: AppProps) {
  const [reveal, setReveal] = useState(false);

  useEffect(() => {setReveal(true)}, []);

  if (!reveal) return <div>Loading...</div>

  return <Component {...pageProps} />
}
```

That'd naturally not include anything from the page (`<Component />`) in the initial HTML render. Do you maybe have something like that?

**Reply by yannick-leguennec** [reply]
Hi @icyJoseph,

Thank you for taking the time to address my issue.

You hit the nail on the head! I indeed had a loader in my `_app` file that was preventing my component tags from rendering properly.

I've moved my logic directly into my component file, and now it works perfectly.

Thank you again.
